editor/src/components/inspector/sections/component-section/component-section.tsx

```diff
@@ -429,18 +424,24 @@ const RowForUnionControl = betterReactMemo(
 
     const simpleControlStyles = getControlStyles('simple')
 
-    const label = (
-      <PopupList
-        value={labelValue}
-        options={labelOptions}
-        onSubmitValue={onLabelChangeValue}
-        containerMode='showBorderOnHover'
-        controlStyles={simpleControlStyles}
-        style={{
-          maxWidth: '100%',
-          overflow: 'hidden',
-        }}
-      />
+    const label = React.useMemo(
```

Comment:
I'm surprised. did eslint tell you there is a problem here? what was the error?

Reply:
labelAsRenderProp has this as a dep, and it would change on every render.

Reply:
I just realized that labelAsRenderProp is a function that returns a function pseudo-component... I uh.. is RowForBaseControl even memoized?

Reply:
sounds like a good ticket